### PR TITLE
Move files to *IT as IntegrationTest is not a standard failsafe suffix

### LINKED ISSUE #25895: com.hazelcast.mapstore.* 'NoClassDefFoundError' [HZ-3660]
_master_ (commit 0d36189619cd30230e4a68818658c2183c427337)
Failed on oracle-11: https://jenkins.hazelcast.com/job/Hazelcast-master-postgres-16/2/testReport/junit/com.hazelcast.mapstore.postgres/

<details><summary>Example Stacktrace:</summary>

```
java.lang.NoClassDefFoundError: com/hazelcast/shaded/org/apache/calcite/rex/RexVisitor
	at java.base/java.lang.Class.getDeclaredConstructors0(Native Method)
	at java.base/java.lang.Class.privateGetDeclaredConstructors(Class.java:3137)
	at java.base/java.lang.Class.getConstructor0(Class.java:3342)
	at java.base/java.lang.Class.getDeclaredConstructor(Class.java:2553)
	at com.hazelcast.internal.util.ServiceLoader$NewInstanceIterator.next(ServiceLoader.java:293)
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
	at com.hazelcast.jet.sql.impl.connector.SqlConnectorCache.<init>(SqlConnectorCache.java:38)
	at com.hazelcast.jet.sql.impl.CalciteSqlOptimizer.<init>(CalciteSqlOptimizer.java:248)
	at com.hazelcast.sql.impl.SqlServiceImpl.start(SqlServiceImpl.java:95)
	at com.hazelcast.spi.impl.NodeEngineImpl.start(NodeEngineImpl.java:292)
	at com.hazelcast.instance.impl.Node.start(Node.java:483)
	at com.hazelcast.instance.impl.HazelcastInstanceImpl.<init>(HazelcastInstanceImpl.java:125)
	at com.hazelcast.instance.impl.HazelcastInstanceFactory.constructHazelcastInstance(HazelcastInstanceFactory.java:218)
	at com.hazelcast.instance.impl.HazelcastInstanceFactory.newHazelcastInstance(HazelcastInstanceFactory.java:197)
	at com.hazelcast.test.TestHazelcastInstanceFactory.newHazelcastInstance(TestHazelcastInstanceFactory.java:236)
	at com.hazelcast.jet.SimpleTestInClusterSupport.initializeWitSupplier(SimpleTestInClusterSupport.java:80)
	at com.hazelcast.jet.SimpleTestInClusterSupport.initialize(SimpleTestInClusterSupport.java:68)
	at com.hazelcast.jet.sql.impl.connector.jdbc.JdbcSqlTestSupport.initialize(JdbcSqlTestSupport.java:78)
	at com.hazelcast.jet.sql.impl.connector.jdbc.JdbcSqlTestSupport.initialize(JdbcSqlTestSupport.java:65)
	at com.hazelcast.mapstore.postgres.PostgresGenericMapStoreTest.beforeClass(PostgresGenericMapStoreTest.java:30)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at com.hazelcast.test.AfterClassesStatement.evaluate(AfterClassesStatement.java:41)
	at com.hazelcast.test.OverridePropertyRule$1.evaluate(OverridePropertyRule.java:66)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.ClassNotFoundException: com.hazelcast.shaded.org.apache.calcite.rex.RexVisitor
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
	... 25 more

```

</details>

<details><summary>Standard output:</summary>

```
11:03:58,970  INFO || - [MetricsConfigHelper] Time-limited test - [LOCAL] [dev] [5.4.0-SNAPSHOT] Overridden metrics configuration with system property 'hazelcast.metrics.collection.frequency'='1' -> 'MetricsConfig.collectionFrequencySeconds'='1'
11:03:58,972  INFO || - [logo] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] 
	+       +  o    o     o     o---o o----o o      o---o     o     o----o o--o--o
	+ +   + +  |    |    / \       /  |      |     /         / \    |         |   
	+ + + + +  o----o   o   o     o   o----o |    o         o   o   o----o    |   
	+ +   + +  |    |  /     \   /    |      |     \       /     \       |    |   
	+       +  o    o o       o o---o o----o o----o o---o o       o o----o    o   
11:03:58,972  INFO || - [system] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Copyright (c) 2008-2023, Hazelcast, Inc. All Rights Reserved.
11:03:58,972  INFO || - [system] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Hazelcast Platform 5.4.0-SNAPSHOT (20231102 - 0d36189) starting at [127.0.0.1]:5701
11:03:58,972  INFO || - [system] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Cluster name: dev
11:03:58,972  INFO || - [system] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Integrity Checker is disabled. Fail-fast on corrupted executables will not be performed. For more information, see the documentation for Integrity Checker.
11:03:58,973  INFO || - [system] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Jet is enabled
11:03:58,983  INFO || - [MetricsConfigHelper] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Collecting debug metrics and sending to diagnostics is enabled
11:03:58,984  INFO || - [TpcServerBootstrap] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] TPC: disabled
11:03:58,998  WARN || - [CPSubsystem] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] CP Subsystem is not enabled. CP data structures will operate in UNSAFE mode! Please note that UNSAFE mode will not provide strong consistency guarantees.
11:03:59,019  INFO || - [JetServiceBackend] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Setting number of cooperative threads and default parallelism to 2
11:03:59,055  WARN || - [Node] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Terminating forcefully...
11:03:59,055  INFO || - [Node] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Shutting down connection manager...
11:03:59,055  INFO || - [Node] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Shutting down node engine...
11:03:59,061  INFO || - [NodeExtension] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Destroying node NodeExtension.
11:03:59,061  INFO || - [Node] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Hazelcast Shutdown is completed in 6 ms.
11:03:59,577  INFO || - [SimpleTestInClusterSupport] Time-limited test - Terminating instance factory in SimpleTestInClusterSupport.@AfterClass

```
</details>

Standard output can be found here - https://s3.console.aws.amazon.com/s3/buckets/j-artifacts/Hazelcast-master-postgres-16/2/

[com.hazelcast.mapstore.postgres.PostgreGenericMapStoreIntegrationTest.(?)](https://jenkins.hazelcast.com/job/Hazelcast-master-postgres-16/lastSuccessfulBuild/testReport/junit/com.hazelcast.mapstore.postgres/PostgreGenericMapStoreIntegrationTest/___/)
[com.hazelcast.mapstore.postgres.PostgresGenericMapStoreTest.(?)](https://jenkins.hazelcast.com/job/Hazelcast-master-postgres-16/lastSuccessfulBuild/testReport/junit/com.hazelcast.mapstore.postgres/PostgresGenericMapStoreTest/___/)
[com.hazelcast.mapstore.mysql.MySQLGenericMapStoreIntegrationTest.(?)](https://jenkins.hazelcast.com/job/Hazelcast-master-mysql-8.2/lastSuccessfulBuild/testReport/junit/com.hazelcast.mapstore.mysql/MySQLGenericMapStoreIntegrationTest/___/)
[com.hazelcast.mapstore.mysql.MySQLGenericMapStoreTest.(?)](https://jenkins.hazelcast.com/job/Hazelcast-master-mysql-8.2/lastSuccessfulBuild/testReport/junit/com.hazelcast.mapstore.mysql/MySQLGenericMapStoreTest/___/)

### PR DESCRIPTION
**Pull request by TomaszGaweda**
Fixes #25895

Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [x] Add `Add to Release Notes` label if changes should be mentioned in release notes or `Not Release Notes content` if changes are not relevant for release notes
- [ ] Request reviewers if possible
- [ ] Send backports/forwardports if fix needs to be applied to past/future releases

### CODE REVIEW

**Review by frant-hartm — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -1315,17 +1315,17 @@
                                 -Dlog4j.skipJansi=true
                                 ${extraVmArgs}
                             </argLine>
+                            <groups>
+                                com.hazelcast.test.annotation.NightlyTest,
+                                com.hazelcast.test.annotation.SlowTest
+                            </groups>
                             <includes>
-                                <include>**/**.java</include>
+                                <include>**/**Test.java</include>
```
Comment (TomaszGaweda) [inline pom.xml:1323]: It was picking tests like PostgreGenericMapStoreIT before that change

**Review by frant-hartm — APPROVED** [review]

File: pom.xml [hunk 2]
```diff
@@ -1345,6 +1345,9 @@
                             </argLine>
                             <useManifestOnlyJar>false</useManifestOnlyJar>
                             <useSystemClassLoader>true</useSystemClassLoader>
+                            <includes>
+                                <include>**/**IT.java</include>
```
Comment (kwart) [inline pom.xml:1349]: Any reason to use the explicit definition?
The default contains the expression too:
```xml
<includes>
    <include>**/IT*.java</include>
    <include>**/*IT.java</include>
    <include>**/*ITCase.java</include>
</includes>
```

File: extensions/mongodb/pom.xml [hunk 3]
```diff
@@ -45,6 +45,7 @@
                             <Automatic-Module-Name>com.hazelcast.jet.mongodb</Automatic-Module-Name>
                         </manifestEntries>
                     </archive>
+                    <forceCreation>true</forceCreation>
```
Comment (kwart) [inline extensions/mongodb/pom.xml:48]: How is this related to the "standard suffix"?

**Review by kwart — COMMENTED** [review]
Is there any plan to move other `IntegrationTest` classes to the `IT` naming?

```bash
find -name \*IntegrationTest\*.java |wc -l
138
```

File: pom.xml (hunk 2 above)
Comment (kwart) [reply pom.xml:1349]: One more thing - the `**` in the filename is unnecessary.

Comment (TomaszGaweda) [reply pom.xml:1349]: Yes, to clearly differentiate between surefire and failsafe. During investigation I had very weird behaviours where some tests were not executed at all and some were executed twice; making config explicit made all tests pass exactly once.
And yes, I used only IT suffix as it's the most commonly used. Should I add the rest of them?

File: extensions/mongodb/pom.xml (hunk 3 above)
Comment (TomaszGaweda) [reply extensions/mongodb/pom.xml:48]: Ah, right. Not strictly related, we just want to force recreating jar files even if nothing changes in given module; we need to shade and relocate some classes, this helps a bit here.

**Comment by TomaszGaweda** [comment]
>Is there any plan to move other IntegrationTest classes to the IT naming?

Step by step, I didn't want to do huge PR with mass change right before feature freeze ;) It's most urgent in cases where order of execution matters, like in tests that rely on shaded (and relocated) classes

File: pom.xml (hunk 2 above)
Comment (kwart) [reply pom.xml:1349]: OK, it make sense to keep it simple - so no additional expressions. I would just use the single star in the file name:
```suggestion
                                <include>**/*IT.java</include>
```

File: extensions/mongodb/pom.xml (hunk 3 above)
Comment (kwart) [reply extensions/mongodb/pom.xml:48]: I would rather see it in a separate PR, but let's leave it here in this case. I'll be more strict next time :grin:

**Comment by kwart** [comment]
> > Is there any plan to move other IntegrationTest classes to the IT naming?
> 
> Step by step, I didn't want to do huge PR with mass change right before feature freeze ;) It's most urgent in cases where order of execution matters, like in tests that rely on shaded (and relocated) classes

If we don't have a continuation issue or JIRA already, could you create one and provide a link here?

**Comment by TomaszGaweda** [comment]
Sure, https://hazelcast.atlassian.net/browse/HZ-3896 @kwart  :)

**Review by kwart — APPROVED** [review]
